src/plugins/intel_gpu/src/graph/fully_connected.cpp

```diff
@@ -165,6 +165,26 @@ std::vector<layout> fully_connected_inst::calc_output_layouts(fully_connected_no
     return { layout{output_shapes[0], output_type, output_format} };
 }
 
+
+kernel_impl_params fully_connected_inst::get_fake_aligned_params(kernel_impl_params const& impl_param) {
+    // fc_tiled_opt kernel is optimized for row shape aligned by 16.
+    // Thus, use fake aligned shape at kernel execution for better performance.
+    if (impl_param.get_input_layout().format == format::bfyx && impl_param.output_layout.format == format::bfyx) {
+        auto fake_param = impl_param;
+        auto input_shape = fake_param.get_input_layout(0).get_partial_shape().to_shape();
+        auto input_row_idx = input_shape.size() - 2;
+        input_shape[input_row_idx] = kernel_selector::Align(input_shape[input_row_idx], 16);
+        auto output_shape = fake_param.output_layout.get_partial_shape().to_shape();
```

Comment:
Don't we need a check that shapes are static and doesn't do anything with params if for dynamic case to avoid exception?